metadata_service/proxy/neo4j_proxy.py

```diff
@@ -1177,3 +1177,46 @@ def put_dashboard_description(self, *,
         self._put_resource_description(resource_type=ResourceType.Dashboard,
                                        uri=id,
                                        description=description)
+
+    @timer_with_counter
+    def get_resources_using_table(self, *,
+                                  id: str,
+                                  resource_type: ResourceType) -> Dict[str, List[DashboardSummary]]:
+        """
+
+        :param id:
+        :param resource_type:
+        :return:
+        """
+        if resource_type != ResourceType.Dashboard:
+            raise NotImplementedError('{} is not supported'.format(resource_type))
+
+        get_dashboards_using_table_query = textwrap.dedent(u"""
+        MATCH (d:Dashboard)-[:DASHBOARD_WITH_TABLE]->(table:Table {key: $query_key}),
+        (d)-[:DASHBOARD_OF]->(dg:Dashboardgroup)-[:DASHBOARD_GROUP_OF]->(c:Cluster)
+        OPTIONAL MATCH (d)-[:DESCRIPTION]->(description:Description)
+        OPTIONAL MATCH (d)-[:EXECUTED]->(last_success_exec:Execution)
+        WHERE split(last_success_exec.key, '/')[5] = '_last_successful_execution'
+        OPTIONAL MATCH (d)-[read:READ_BY]->(:User)
+        WITH c, dg, d, description, last_success_exec, sum(read.read_count) as recent_view_count
+        RETURN
+        d.key as uri,
+        c.name as cluster,
+        dg.name as group_name,
+        dg.dashboard_group_url as group_url,
+        d.name as name,
+        d.dashboard_url as url,
+        description.description as description,
+        split(d.key, '_')[0] as product,
```

Comment:
I forgot, but do we have product in any node/ attribute of the dashboard node? any reason we need to do a split to get the product instead of directly fetch the node?

Reply:
Yeah, we actually don't have it currently.

May be we can have product node just like Database node later on?

Reply:
yeah, having a product node seems better.